ext/compressors/lz4/lz4_compress.c

```diff
@@ -48,19 +48,41 @@ typedef struct {
 } LZ4_COMPRESSOR;
 
 /*
+ * LZ4 decompression requires the exact compressed byte count returned by the
+ * LZ4_compress and LZ4_compress_destSize functions. WiredTiger doesn't track
+ * that value, store it in the destination buffer.
+ *
+ * Additionally, LZ4_compress_destSize may compress into the middle of a record,
+ * and after decompression we return the length to the last record successfully
+ * decompressed, not the number of bytes decompressed; store that value in the
+ * destination buffer as well.
+ *
+ * Use fixed-size, 4B values (WiredTiger never writes buffers larger than 4GB).
+ *
+ * The unused field is available for a mode flag if one is needed in the future,
+ * we guarantee it's 0.
+ */
+typedef struct {
+	uint32_t compressed_len;	/* True compressed length */
+	uint32_t uncompressed_len;	/* True uncompressed source length */
+	uint32_t useful_len;		/* Decompression return value */
+	uint32_t unused;		/* Guaranteed to be 0 */
```

Comment:
We could switch the order, sure, I don't think it makes much difference.

@michaelcahill pointed out we could do versioning by changing the compressor name (for example, the next version could be `lz4-number-2` or something like that). That means we wouldn't need to version the block itself, unless for some reason we wanted to mix-and-match LZ4 compression versions in the same file.